src/Buffer.h

```diff
@@ -2,144 +2,107 @@
 #define HALIDE_BUFFER_H
 
 /** \file
- * Defines Buffer - A c++ wrapper around a buffer_t.
+ * Defines BufferPtr - A c++ wrapper around a buffer_t.
```

Comment:
The filename is now out of convention -- should we change it to BufferPtr.h (in a following PR)?